src/typ/pattern.rs

```diff
@@ -399,6 +399,7 @@ impl<'a, 'b, 'c> PatternTyper<'a, 'b, 'c> {
                     ValueConstructorVariant::Record { ref name, .. } => {
                         PatternConstructor::Record { name: name.clone() }
                     }
+                    ValueConstructorVariant::Inline => todo!(),
```

Comment:
Here I think we would need to add a new `PatternConstructor` variant called `Inline` to match the value constructor.